### PR TITLE
Perl linter

### PR DESCRIPTION
**Pull request by jreidinger**
Is that check_source.pl used at all? Some bugs in it looks bad enough, especially that regexp.

### CODE REVIEW

**Comment by jreidinger** [comment]
Too make this work it needs to adapt test docker container to contain also all required perl libraries.

**Comment by coolo** [comment]
Yes, check_source.pl is used by check_source.py (just to add to the fun of it :)

And that regexp bug isn't as bad as it first looks - it just means we also look for patches mentioned in other parts of the diff, not just the + part. I guess this leads to 0 problems in reality - but good catch anyway.

**Comment by coolo** [comment]
And the CI container is defined in `dist/ci/testenv-tumbleweed/Dockerfile` - but I just learned it's broken:
https://build.opensuse.org/package/show/openSUSE:Tools:Images/osrt-testenv-tumbleweed

**Comment by jreidinger** [comment]
@coolo thanks, I will check it and try fix that container.

**Comment by coolo** [comment]
Fixed the container in OBS - the problem was an inconsistent file provides between Factory and openSUSE:Tools:Images

File: .perl_checker.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#! /usr/bin/sh
+for i in `find . | grep 'p[lm]$'`; do
```
Comment (coolo) [inline .perl_checker.sh:2]: please no hidden files as commands. We have dist/ci for that

**Review by coolo — CHANGES_REQUESTED** [review]

**Review by coolo — APPROVED** [review]

**Comment by DimStar77** [comment]
> Yes, check_source.pl is used by check_source.py (just to add to the fun of it :)
> 
> And that regexp bug isn't as bad as it first looks - it just means we also look for patches mentioned in other parts of the diff, not just the + part. I guess this leads to 0 problems in reality - but good catch anyway.

The SLE submissions actually relied on this bug

i.e:
* a patch is added in SLE-SP-(n-1), referenced in the changelog, as is needed.
* in SLE-SP-n, the package is taken from Factory, which never had the patch. The diff now shows the patch being 'lost', which was considered ok (the changelog never mentions the patch being added, so we also do not drop it in the changelog)
* So, the fix is technically seen correct, but causes more issues for SLE now - as patches can't formally be 'lost' anymore

**Comment by DimStar77** [comment]
The above note is also a problem for TW-proper: if I have to submit a 'revert submission SR', then patches are being dropped the same way (i.e. never added) - the bot would now decline those too

**Comment by coolo** [comment]
Well, then this should be properly implemented not rely on a bug

**Comment by DimStar77** [comment]
> Well, then this should be properly implemented not rely on a bug

Of course - I wasn't trying to imply we should put the bug back in place. Just that the bug was unnoticed for a long time and 'served us well'

**Comment by suntorytimed** [comment]
Agreeing with DimStar77 and coolo, we should not rely on a bug for this. Nevertheless we require a proper implementation sooner than later, as this will hit us with a couple of submissions to SLE. I have opened a bug and linked it to this PR.

**Pull request merged**